xclim/sdba/_adjustment.py

```diff
@@ -615,3 +615,335 @@ def extremes_adjust(
+    """
+    hist = ds.hist.stack(dim_hist=dim)
+    ref = ds.ref.stack(dim_ref=dim, create_index=False)
+    hist = hist.dropna(dim="dim_hist")
+    ref = ref.dropna(dim="dim_ref")
+
+    scen = xr.apply_ufunc(
+        _otc_adjust,
+        hist,
+        ref,
+        kwargs=dict(
+            bin_width=bin_width,
+            bin_origin=bin_origin,
+            num_iter_max=num_iter_max,
+            spray_bins=spray_bins,
+        ),
+        input_core_dims=[["dim_hist", "multivar"], ["dim_ref", "multivar"]],
+        output_core_dims=[["dim_hist", "multivar"]],
+        keep_attrs=True,
+        vectorize=True,
+    )
+
+    scen = scen.unstack().rename("scen")
+
+    return scen.to_dataset()
+
+
+def _dotc_adjust(
+    X1: np.ndarray,
+    Y0: np.ndarray,
+    X0: np.ndarray,
+    bin_width: list | None = None,
+    bin_origin: list | None = None,
+    num_iter_max: int | None = 100_000_000,
+    cov_factor: str | None = "std",
+    spray_bins: bool = True,
+    kind: dict | None = None,
+):
+    """Dynamical Optimal Transport Correction of the bias of X with respect to Y.
+
+    Parameters
+    ----------
+    X1 : np.ndarray
+        Simulation data to adjust.
+    Y0 : np.ndarray
+        Bias correction reference.
+    X0 : np.ndarray
+        Historical simulation data.
+    bin_width : list | None
+        Bin widths for all dimensions.
+    bin_origin : list | None
+        Bin origins for all dimensions.
+    num_iter_max : int | None
+        Maximum number of iterations used in the earth mover distance algorithm.
+    cov_factor : str | None = "std"
+        Rescaling factor.
+
+    Returns
+    -------
+    np.ndarray
+        Adjusted data
+
+    References
+    ----------
+    :cite:cts:`sdba-robin_2021`
+    """
+    # Initialize parameters
+    bin_width = u.bin_width_estimator([Y0, X0, X1]) if bin_width is None else bin_width
+    if cov_factor == "cholesky":
+        fact0 = u.eps_cholesky(np.cov(Y0, rowvar=False))
+        fact1 = u.eps_cholesky(np.cov(X0, rowvar=False))
+        cov_factor = np.dot(fact0, np.linalg.inv(fact1))
+    elif cov_factor == "std":
+        fact0 = np.std(Y0, axis=0)
+        fact1 = np.std(X0, axis=0)
+        cov_factor = np.diag(fact0 / fact1)
+    else:
+        cov_factor = np.identity(Y0.shape[1])
+
+    # Map ref to hist
+    yX0 = _otc_adjust(
+        Y0,
+        X0,
+        bin_width=bin_width,
+        bin_origin=bin_origin,
+        num_iter_max=num_iter_max,
+        spray_bins=False,
+    )
+
+    # Map hist to sim
+    yX1 = _otc_adjust(
+        X0,
+        X1,
+        src=yX0,
+        bin_width=bin_width,
+        bin_origin=bin_origin,
+        num_iter_max=num_iter_max,
+        spray_bins=False,
+    )
+
+    # Temporal evolution
+    motion = np.empty(yX0.shape)
+    for j in range(yX0.shape[1]):
+        if kind is not None and kind[j] == "*":
+            motion[:, j] = yX1[:, j] / yX0[:, j]
+        else:
+            motion[:, j] = yX1[:, j] - yX0[:, j]
```

Comment:
`input_core_dims` was `[dim_ref, "multivar" ]` previously, now it seems 'multivar' is the first coordinate. Is there a transpose somewhere or I'm not understanding something?